src/views/SignUp/SignUp.js

```diff
@@ -0,0 +1,107 @@
+import React,{Component} from "react";
+import Layout from '../../components/Layout';
+import camelcase from 'camelcase';
+import serverApi from '../../api/worker';
+
+export default class SignUp extends Component{
+
+  constructor(props){
+    super(props);
+
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      username:'',
+      password:'',
+      verifyPassword:''
+    };
+  }
+
+  fields = [
+    //["label", "Type", Validation method]
+    ["First name","text", v => v.length > 0],
+    ["Last name","text", v => v.length > 0],
+    ["Email","email", v => (/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/).test(v)], // eslint-disable-line],
+    ["Username","text", v => v.length > 0],
+    ["Password","password", v => v.length >= 8],
+    ["Verify password","password", v => v === this.state.password && v.length>=8]
+  ]
+ /**
+   * event listener for <input/>'s
+   * adds changes to this.state
+   * @param  {event}
```

Comment:
Its the type which should be described within `{}`
```suggestion
   * @param  {Object} event
```